src/allocators/ERC7683Allocator.sol

```diff
@@ -0,0 +1,420 @@
+        0x037a34e1ded3bcc84f59dfc185efc3553c509ebab317153a8dddefce2eaee6f0;
+
+    /// @notice The typehash of the OrderDataGasless struct
+    //          keccak256("OrderDataGasless(Order order,bool deposit)
+    //          Lock(bytes12 lockTag,address token,uint256 amount)
+    //          Order(address arbiter,Lock[] commitments,uint256 chainId,address tribunal,address recipient,address settlementToken,uint256 minimumAmount,uint256 baselinePriorityFee,uint256 scalingFactor,uint256[] decayCurve,bytes32 salt,bytes32 qualification)")
+    bytes32 public constant ORDERDATA_GASLESS_TYPEHASH =
+        0x79e4af6feaa84a46fd69ed25e4595e9f6e8690ba3a6c564bfa235542f9faf55c;
+
+    /// @notice keccak256("BatchCompact(address arbiter,address sponsor,uint256 nonce,uint256 expires,Lock[] commitments,Mandate mandate)
+    //          Lock(bytes12 lockTag,address token,uint256 amount)
+    //          Mandate(uint256 chainId,address tribunal,address recipient,uint256 expires,address token,uint256 minimumAmount,uint256 baselinePriorityFee,uint256 scalingFactor,uint256[] decayCurve,bytes32 salt)")
+    bytes32 public constant BATCH_COMPACT_WITNESS_TYPEHASH =
+        0x5ede122c736b60a8b718f83dcfb5d6e4aa27c9714d0c7bc9ca86562b8f878463;
+
+    /// @notice keccak256("Mandate(uint256 chainId,address tribunal,address recipient,uint256 expires,address token,uint256 minimumAmount,uint256 baselinePriorityFee,uint256 scalingFactor,uint256[] decayCurve,bytes32 salt)")
+    bytes32 private constant MANDATE_TYPEHASH = 0x74d9c10530859952346f3e046aa2981a24bb7524b8394eb45a9deddced9d6501;
+
+    mapping(bytes32 claimHash => bytes32 qualification) public qualifications;
+
+    constructor(address compactContract_) OnChainAllocator(compactContract_) {}
+
+    /// @inheritdoc IOriginSettler
+    function openFor(GaslessCrossChainOrder calldata order_, bytes calldata sponsorSignature_, bytes calldata)
+        external
+    {
+        // Check if orderDataType is the one expected by the allocator
+        if (order_.orderDataType != ORDERDATA_GASLESS_TYPEHASH) {
+            revert InvalidOrderDataType(order_.orderDataType, ORDERDATA_GASLESS_TYPEHASH);
+        }
+        if (order_.originSettler != address(this)) {
+            revert InvalidOriginSettler(order_.originSettler, address(this));
+        }
+
+        // Decode the orderData
+        (Order calldata orderData, uint32 deposit) = _decodeOrderData(order_.orderData);
+
+        uint160 caller = uint160(deposit * uint160(msg.sender)); // for a deposit, the nonce will be scoped to the caller + user
+        bytes32 nonceIdentifier = _toNonceId(address(caller), order_.user);
+
+        // Early revert if the expected nonce is not the next nonce
+        if (order_.nonce != nonces[nonceIdentifier] + 1) {
+            revert InvalidNonce(order_.nonce, nonces[nonceIdentifier] + 1);
+        }
+
+        bytes32 qualification = bytes32(uint256(orderData.qualification) * deposit); // delete qualification if not a deposit
```

Comment:
comment could be more clear I think - ie why? maybe like "branchless: if not deposit => 0, else qualification * deposit" or something